packages/amplify-cli/src/execution-manager.ts

```diff
@@ -245,19 +254,28 @@ async function raisePrePullEvent(context: Context) {
   await raiseEvent(context, new AmplifyEventArgs(AmplifyEvent.PrePull, new AmplifyPrePullEventData()));
 }
 
+async function raisePreCodegenModelsEvent(context: Context) {
+  await raiseEvent(context, new AmplifyEventArgs(AmplifyEvent.PreCodegenModels, new AmplifyPreCodegenModelsEventData()));
+}
+
 async function raisePostEvent(context: Context) {
-  if (context.input.plugin === constants.CORE) {
-    switch (context.input.command) {
-      case 'init':
-        await raisePostInitEvent(context);
-        break;
-      case 'push':
-        await raisePostPushEvent(context);
-        break;
-      case 'pull':
-        await raisePostPullEvent(context);
-        break;
-    }
+  const { command, plugin } = context.input;
+  if (!plugin || !PLUGINS_WITH_HANDLERS.has(plugin)) {
```

Comment:
can just check that the plugin is in the set. Since the set doesn't contain undefined you automatically get the first check